Player.cpp

```diff
@@ -0,0 +1,37 @@
+#include <iostream>
+#include <string>
+#include "player.h"
+
+Player::Player(string name, int HP, int DMG) : Name(name), Hp(HP), Dmg(DMG) {}
+
+string Player::getName() const
+{
+	return Name;
+}
+int Player::getHp() const
+{
+	return Hp;
+}
+int Player::getDmg() const
+{
+	return Dmg;
+}
+
+void Player::atck(Player& target) const {
+	cout << this->Name << " -> " << target.Name << endl;
+
+	if (target.Hp < this->Dmg) {
+		target.Hp = 0;
+	}
+	else {
+		target.Hp -= this->Dmg;
```

Comment:
A `this`ek feleslegesek. A `Name` az az aktuális objektum neve, aminek a metódusát meghívták.